streams/src/test/java/org/apache/kafka/streams/integration/KStreamCogroupIntegrationTest.java

```diff
@@ -0,0 +1,489 @@
+        stream1.cogroup(AGGREGATOR_1)
+                .cogroup(stream2, AGGREGATOR_2)
+                .cogroup(stream3, AGGREGATOR_3)
+                .aggregate(INITIALIZER, null, COGROUP_STORE_NAME)
+                .to(OUTPUT_TOPIC + testNumber);
+
+        final KafkaStreams streams = new KafkaStreams(builder, streamsConfig(APP_ID + testNumber));
+        
+        final List<KeyValue<Long, String>> expecteds = Arrays.asList(
+                KeyValue.pair(1L, "1a"),
+                KeyValue.pair(2L, "2a"),
+                KeyValue.pair(1L, "1a3a"),
+                KeyValue.pair(1L, "1a3a1b"),
+                KeyValue.pair(2L, "2a2b"),
+                KeyValue.pair(1L, "1a3a1b3b"),
+                KeyValue.pair(2L, "2a2b1c"),
+                KeyValue.pair(1L, "1a3a1b3b2c"),
+                KeyValue.pair(2L, "2a2b1c3c"),
+                KeyValue.pair(2L, "2a2b1c3c1a"),
+                KeyValue.pair(1L, "1a3a1b3b2c2a"),
+                KeyValue.pair(2L, "2a2b1c3c1a3a"),
+                KeyValue.pair(2L, "2a2b1c3c1a3a1b"),
+                KeyValue.pair(1L, "1a3a1b3b2c2a2b"),
+                KeyValue.pair(2L, "2a2b1c3c1a3a1b3b"),
+                KeyValue.pair(1L, "1a3a1b3b2c2a2b1c"),
+                KeyValue.pair(2L, "2a2b1c3c1a3a1b3b2c"),
+                KeyValue.pair(1L, "1a3a1b3b2c2a2b1c3c")
+            );
+
+        try {
+            streams.start();
+
+            final Iterator<KeyValue<Long, String>> expectedsIterator = expecteds.iterator();
+            for (final Input<Long, String> input : INPUTS) {
+                IntegrationTestUtils.produceKeyValuesSynchronously(input.topic + testNumber, Collections.singleton(input.keyValue), producerConfig, CLUSTER.time);
+                List<KeyValue<Long, String>> outputs = IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived(consumerConfig, OUTPUT_TOPIC + testNumber, 1);
+                assertThat(outputs.get(0), equalTo(expectedsIterator.next()));
+            }
+        } finally {
+            streams.close();
+        }
+    }
+
+    @Test
+    public void testCogroupSessionWindow() throws InterruptedException, ExecutionException {
+        final int testNumber = TEST_NUMBER.getAndIncrement();
+        final Properties producerConfig = producerConfig();
+        final Properties consumerConfig = consumerConfig("consumer-" + testNumber);
+        consumerConfig.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, WindowedDeserializer.class);
+        consumerConfig.put("key.deserializer.inner.class", LongDeserializer.class.getName());
+        CLUSTER.createTopic(INPUT_TOPIC_1 + testNumber, 2, 1);
+        CLUSTER.createTopic(INPUT_TOPIC_2 + testNumber, 2, 1);
+        CLUSTER.createTopic(INPUT_TOPIC_3 + testNumber, 2, 1);
+        CLUSTER.createTopic(OUTPUT_TOPIC + testNumber, 2, 1);
+
+        final long timestamp = CLUSTER.time.milliseconds();
+
+        final KStreamBuilder builder = new KStreamBuilder();
```

Comment:
nit: i don't mind if you use `final` or not in the methods, but it would be good to be consistent.

Same in other tests